src/main/scala/com/typesafe/sbt/packager/docker/DockerPlugin.scala

```diff
@@ -153,16 +187,22 @@ object DockerPlugin extends AutoPlugin {
       stagingDirectory := (target in Docker).value / "stage",
       target := target.value / "docker",
       daemonUser := "daemon",
-      daemonGroup := daemonUser.value,
+      daemonGroup := "root",
       defaultLinuxInstallLocation := "/opt/docker",
+      validatePackage := Validation
+        .runAndThrow(validatePackageValidators.value, streams.value.log),
       validatePackageValidators := Seq(
         nonEmptyMappings((mappings in Docker).value),
         filesExist((mappings in Docker).value),
         validateExposedPorts(dockerExposedPorts.value, dockerExposedUdpPorts.value),
-        validateDockerVersion(dockerVersion.value)
+        validateDockerVersion(dockerVersion.value),
+        validateDockerPermissionStrategy(dockerPermissionStrategy.value, dockerVersion.value)
       ),
       dockerPackageMappings := MappingsHelper.contentOf(sourceDirectory.value),
-      dockerGenerateConfig := generateDockerConfig(dockerCommands.value, stagingDirectory.value)
+      dockerGenerateConfig := {
+        val _ = validatePackage.value
```

Comment:
What are your thoughts on this pattern? I usually tend to do something like

```sbt
dockerGenerateConfig := { /* implementation */ },
// side effects
dockerGenerateConfig := (dockerGenerateConfig dependsOn validatePackage).value
```

Reply:
I think of `dependsOn` as part of sbt 0.12 style used together with `<<=` etc, since with `.value` you can just call it like I did.

Reply:
Thanks for sharing :smile: